stdlib/public/Darwin/Accelerate/vDSP_FillClearGenerate.swift

```diff
@@ -0,0 +1,416 @@
+                switch type {
+                case .hanningNormalized:
+                    vDSP_hann_windowD(v.baseAddress!,
+                                      n,
+                                      Int32(vDSP_HANN_NORM) |
+                                        Int32(isHalfWindow ?  vDSP_HALF_WINDOW : 0))
+                case .hanningDenormalized:
+                    vDSP_hann_windowD(v.baseAddress!,
+                                      n,
+                                      Int32(vDSP_HANN_DENORM) |
+                                        Int32(isHalfWindow ?  vDSP_HALF_WINDOW : 0))
+                case .hamming:
+                    vDSP_hamm_windowD(v.baseAddress!,
+                                      n,
+                                      Int32(isHalfWindow ?  vDSP_HALF_WINDOW : 0))
+                case .blackman:
+                    vDSP_blkman_windowD(v.baseAddress!,
+                                        n,
+                                        Int32(isHalfWindow ?  vDSP_HALF_WINDOW : 0))
+                }
+            }
+    }
+    
+    // MARK: Ramps
+    
+    /// Fills a supplied array with monotonically incrementing or decrementing values, single-precision.
+    ///
+    /// - Parameter initialValue: Specifies the initial value.
+    /// - Parameter increment: The increment (or decrement if negative) between consecutive elements.
+    /// - Parameter result: Output values.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func generateRamp<V>(withInitialValue initialValue: Float,
+                                       increment: Float,
+                                       result: inout V)
+        where V: _MutableContiguousCollection,
+        V.Element == Float {
+            
+            let n = vDSP_Length(result.count)
+            
+            result.withUnsafeMutableBufferPointer { v in
+                vDSP_vramp([initialValue],
+                           [increment],
+                           v.baseAddress!, 1,
+                           n)
+            }
+    }
+    
+    /// Fills a supplied array with monotonically incrementing or decrementing values, double-precision.
+    ///
+    /// - Parameter initialValue: Specifies the initial value.
+    /// - Parameter increment: The increment (or decrement if negative) between consecutive elements.
+    /// - Parameter result: Output values.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func generateRamp<V>(withInitialValue initialValue: Double,
+                                       increment: Double,
+                                       result: inout V)
+        where V: _MutableContiguousCollection,
+        V.Element == Double {
+            
+            let n = vDSP_Length(result.count)
+            
+            result.withUnsafeMutableBufferPointer { v in
+                vDSP_vrampD([initialValue],
```

Comment:
It occurred to me, that maybe it is possible to make `CollectionOfOne` conform to the `_ContiguouslyStored` and avoid a singleton array allocation here.

Reply:
And... we can't. At least not that easily. `CollectionOfOne._element` is `internal`.

Reply:
We don't actually need to construct a singleton array here, right? Just a pointer to initialValue.

Reply:
@stephentyrone that would work too.

Reply:
I've used a single element array as a shortcut, let me know if you think a pointer would be better form.

Reply:
A pointer would be a minor efficiency win. It's also just a bit cleaner.

Reply:
Should just be able to write:

```swift
var mutableTmp = initialValue
...
vDSP_vramp(&mutableTmp,
```

Reply:
That does work, but it feels (at least to me) sort of clunky - we're not really mutating anything, just working round the vDSP API. Over in my [clipping PR](https://github.com/apple/swift/pull/23231/files), I use `withUnsafePointer`, I'll take your lead on a standardized approach.

Reply:
The `var` here means less mutation and more guaranteed materialization. Or that's how I rationalize it.